kafka-connect-s3/src/main/java/io/confluent/connect/s3/auth/AwsAssumeRoleCredentialsProvider.java

```diff
@@ -77,32 +81,37 @@ public void configure(Map<String, ?> configs) {
     final String secretKey = (String) configs.get(AWS_SECRET_ACCESS_KEY_CONFIG);
     if (StringUtils.isNotBlank(accessKeyId) && StringUtils.isNotBlank(secretKey)) {
       basicCredentials = new BasicAWSCredentials(accessKeyId, secretKey);
+      stsCredentialProvider = new STSAssumeRoleSessionCredentialsProvider
+          .Builder(roleArn, roleSessionName)
+          .withStsClient(AWSSecurityTokenServiceClientBuilder
+              .standard()
+              .withCredentials(new AWSStaticCredentialsProvider(basicCredentials)).build()
+          )
+          .withExternalId(roleExternalId)
+          .build();
     } else {
       basicCredentials = null;
+      stsCredentialProvider = new STSAssumeRoleSessionCredentialsProvider
+          .Builder(roleArn, roleSessionName)
+          // default sts client will internally use default credentials chain provider
+          // https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+          .withStsClient(AWSSecurityTokenServiceClientBuilder.defaultClient())
+          .withExternalId(roleExternalId)
+          .build();
     }
   }
 
   @Override
   public AWSCredentials getCredentials() {
-    if (basicCredentials != null) {
-      return new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, roleSessionName)
-              .withStsClient(AWSSecurityTokenServiceClientBuilder.standard()
-                 .withCredentials(new AWSStaticCredentialsProvider(basicCredentials)).build())
-              .withExternalId(roleExternalId)
-              .build()
-              .getCredentials();
-    } else {
-      return new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, roleSessionName)
-            .withStsClient(AWSSecurityTokenServiceClientBuilder.defaultClient())
-            .withExternalId(roleExternalId)
-            .build()
-            .getCredentials();
-    }
+    return stsCredentialProvider.getCredentials();
   }
 
   @Override
   public void refresh() {
-    // Nothing to do really, since we acquire a new session every getCredentials() call.
+    // performs a force refresh of credentials
```

Comment:
Do we also need to refresh basicCredential also?

Reply:
No we don't need to refresh that, since basicCredentials if set (i.e provided in connector config) are not ephemeral in this case. 

If basic credentials was null, the sts credential provider would use default credentials provider chain that should automatically refresh the creds provided from underlying host...